### PR TITLE
Updated timeline slider component to automatically fetch date range for a given variable

### PR DESCRIPTION
**Pull request by dwnoble**
Screenshots:

## Most recent data available:
slider-recent

## Historical data
slider-1933

## Error initializing component
slider-error

### CODE REVIEW

File: static/library/slider_component.ts [hunk 1]
```diff
@@ -45,42 +70,100 @@ import { ChartEventDetail } from "../js/chart/types";
...
+   * Defaults to https://datacommons.org
    */
-  @property({ type: Number })
-  max: number;
+  @property()
+  apiRoot: string;
 
   /**
-   * Minimum slider value
+   * Type of child place to rank (ex: State, County)
```
Comment (juliawu) [inline static/library/slider_component.ts:145]: Remove "to rank".

File: static/library/slider_component.ts [hunk 2]
```diff
@@ -15,9 +15,34 @@
...
 import { css, CSSResult, html, LitElement, TemplateResult } from "lit";
-import { customElement, property } from "lit/decorators.js";
+import { customElement, property, state } from "lit/decorators.js";
 
 import { ChartEventDetail } from "../js/chart/types";
+import { DEFAULT_API_ENDPOINT } from "./constants";
+import { convertArrayAttribute } from "./utils";
+
+const MOST_RECENT_TEXT = "Most recent";
+const MOST_RECENT_TEXT_LONG = "Most recently available";
```
Comment (juliawu) [inline static/library/slider_component.ts:25]: I don't see where `MOST_RECENT_TEXT_LONG` gets used?

**Review by juliawu — APPROVED** [review]
Nice!

File: static/library/slider_component.ts (hunk 1 above)
Comment (dwnoble) [reply static/library/slider_component.ts:145]: Fixed!

File: static/library/slider_component.ts (hunk 2 above)
Comment (dwnoble) [reply static/library/slider_component.ts:25]: Good catch-removed!

**Pull request merged**